test_spack.py

```diff
@@ -27,10 +28,15 @@ def Run(self, command: str, cwd='.', parallel=False):
         if parallel:
             if machine == 'tsa':
                 srun = 'srun -c 16 -t 01:00:00'
+            if machine == 'daint':
+                srun = 'srun -C gpu -A g110 -t 01:00:00'
+
+        # randomly delay start of installation to avoid write-locks
+        delay = randint(5, 20)
 
         # 2>&1 redirects stderr to stdout
         subprocess.run(
-            f'{setup} (cd {cwd} ; {srun} {command}) >> {machine}_{self.package_name}_{self._testMethodName}.log 2>&1',
+            f'{setup} (cd {cwd} ; {srun} sleep {delay} && {command}) >> {machine}_{self.package_name}_{self._testMethodName}.log 2>&1',
```

Comment:
Never change a running system...

Better keep it. Especially daints file-system latency varies day to day.

Reply:
But adding `sleep {delay} &&` IS the change ;-)

Reply:
I feel like sleep is a workaround for spack's inability to deal with daint's file system. If it's not strictly required, I think we should do it without sleep, otherwise we might have a workaround to a problem that is already solved with the better db_lock_timeout.

Reply:
The tests show, it is strictly needed.